firmware/ui/inc/config_storage.hh

```diff
@@ -0,0 +1,309 @@
+#pragma once
+
+#include "app_main.hh"
+#include "font.hh"
+#include "logger.hh"
+#include "m24c02_eeprom.hh"
+#include <cstring>
+
+// TODO split into header and source files
+class ConfigStorage
+{
+public:
+    // Permanent addresses
+    // NOTE - this is the order in which the settings are saved to the eeprom
+    enum class Config_Id
+    {
```

Comment:
You should put a config format version in here. This will allow you to later change the config format, read the first byte and tell if it is old format or new